### PR TITLE
Dataviz: Improve chart

### PR DESCRIPTION
**Pull request by tkohr**
PR
- adds a background and border to the chart
- gives it a fixed height (like map and table) and improves positioning of its elements
- truncates long axis labels to prevent breaking charts
- now also displays loading mask, when loading data

### CODE REVIEW

**Comment by jahow** [comment]
> Note: A potential further improvement could be to add a loader when changing chart params.

Definitely! can it be part of this PR?

**Comment by tkohr** [comment]
> Definitely! can it be part of this PR?

Yes, I added a commit that fixes the loading mask, which has already been implemented for loading data. Once the data is present, generating new charts by changing its parameters seems to be so fast (even with big datasets) that the loader doesn't need to be triggered here, I have the impression.

**Comment by fgravin** [comment]
UX experts say that if the loading time is < to 1sec, then we shouldn't overload the informations with a spinner.
¯\_(ツ)_/¯

**Comment by tkohr** [comment]
> UX experts say that if the loading time is < to 1sec, then we shouldn't overload the informations with a spinner. `¯\_(ツ)_/¯`

Ok, this confirms to trigger the loader when loading data (like in table and map) and not when changing chart params.

File: libs/ui/dataviz/src/lib/chart/chart.component.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ import {
   ScatterController,
   Tooltip,
 } from 'chart.js'
+import { truncateString } from '../utils/utils'
```
Comment (jahow) [inline libs/ui/dataviz/src/lib/chart/chart.component.ts:30]: I don't think this is really necessary to introduce a new function... this util does exactly the same thing as `String.substring()`

**Review by jahow — CHANGES_REQUESTED** [review]
Thanks, it looks much better! I had an error when loading http://localhost:4200/dataset/c1537a99-733a-49c5-b838-b4f4ca5a3e67 and using "gdb_geomattr_data" as X axis.
image.

I think it's because the `getLabelForValue` function from chart.js might return null sometimes.

File: libs/ui/dataviz/src/lib/chart/chart.component.ts (hunk 1 above)
Comment (tkohr) [reply libs/ui/dataviz/src/lib/chart/chart.component.ts:30]: I guess you are right, you'd just truncate without ending the string with `...` then?

**Comment by tkohr** [comment]
Thanks for finding the error, which indeed seems to occur when the selected data column has no data. I'll add a `?` for that.

File: libs/ui/dataviz/src/lib/chart/chart.component.ts (hunk 1 above)
Comment (jahow) [reply libs/ui/dataviz/src/lib/chart/chart.component.ts:30]: oh right sorry, I overlooked the '...' added at the end! still, I think adding a new `utils/utils.ts` file is a bit awkward, I mean this function could just be part of `chart.component.ts` maybe? just to remove a bit of verbosity

Comment (jahow) [reply libs/ui/dataviz/src/lib/chart/chart.component.ts:30]: the thing is it's quite specific to chart.js because usually this would be handled in CSS

Comment (tkohr) [reply libs/ui/dataviz/src/lib/chart/chart.component.ts]: Ok, I think I got your point. I added a commit to move it to the component. Just wondering if handling the scope in the callback could be done in a more elegant way (to access both the function and the parent scope).

**Review by jahow — APPROVED** [review]

**Pull request merged**